base-notebook/Dockerfile

```diff
@@ -80,72 +82,61 @@ RUN echo "auth requisite pam_deny.so" >> /etc/pam.d/su && \
-ARG CONDA_MIRROR=https://github.com/conda-forge/miniforge/releases/latest/download
-
-# ---- Miniforge installer ----
-# Check https://github.com/conda-forge/miniforge/releases
-# Package Manager and Python implementation to use (https://github.com/conda-forge/miniforge)
-# We're using Mambaforge installer, possible options:
-# - conda only: either Miniforge3 to use Python or Miniforge-pypy3 to use PyPy
-# - conda + mamba: either Mambaforge to use Python or Mambaforge-pypy3 to use PyPy
-# Installation: conda, mamba, pip
-RUN set -x && \
-    # Miniforge installer
-    miniforge_arch=$(uname -m) && \
-    miniforge_installer="Mambaforge-Linux-${miniforge_arch}.sh" && \
-    wget --quiet "${CONDA_MIRROR}/${miniforge_installer}" && \
-    /bin/bash "${miniforge_installer}" -f -b -p "${CONDA_DIR}" && \
-    rm "${miniforge_installer}" && \
-    # Conda configuration see https://conda.io/projects/conda/en/latest/configuration.html
-    conda config --system --set auto_update_conda false && \
-    conda config --system --set show_channel_urls true && \
-    if [[ "${PYTHON_VERSION}" != "default" ]]; then mamba install --quiet --yes python="${PYTHON_VERSION}"; fi && \
-    # Pin major.minor version of python
-    mamba list python | grep '^python ' | tr -s ' ' | cut -d ' ' -f 1,2 >> "${CONDA_DIR}/conda-meta/pinned" && \
-    # Using conda to update all packages: https://github.com/mamba-org/mamba/issues/1092
-    conda update --all --quiet --yes && \
-    conda clean --all -f -y && \
-    rm -rf "/home/${NB_USER}/.cache/yarn" && \
-    fix-permissions "${CONDA_DIR}" && \
-    fix-permissions "/home/${NB_USER}"
-
-# Using fixed version of mamba in arm, because the latest one has problems with arm under qemu
-# See: https://github.com/jupyter/docker-stacks/issues/1539
-RUN set -x && \
-    arch=$(uname -m) && \
-    if [ "${arch}" == "aarch64" ]; then \
-        mamba install --quiet --yes \
-            'mamba<0.18' && \
-            mamba clean --all -f -y && \
-            fix-permissions "${CONDA_DIR}" && \
-            fix-permissions "/home/${NB_USER}"; \
-    fi;
-
-# Install Jupyter Notebook, Lab, and Hub
+# Download and install Micromamba, and initialize Conda prefix.
+#   <https://github.com/mamba-org/mamba#micromamba>
+#   Similar projects using Micromamba:
+#     - Micromamba-Docker: <https://github.com/mamba-org/micromamba-docker>
+#     - repo2docker: <https://github.com/jupyterhub/repo2docker>
+# Install Python, Mamba, Jupyter Notebook, Lab, and Hub
 # Generate a notebook server config
-# Cleanup temporary files
+# Cleanup temporary files and remove Micromamba
 # Correct permissions
 # Do all this in a single RUN command to avoid duplicating all of the
 # files across image layers when the permissions change
-RUN mamba install --quiet --yes \
-    'notebook' \
-    'jupyterhub' \
-    'jupyterlab' && \
+COPY --chown="${NB_UID}:${NB_GID}" initial-condarc "${CONDA_DIR}/.condarc"
```

Comment:
I don't remember clearly, but I recall that `--chown` wasn't supported or behaved a bit differently - watch out for possible issues about that. I'm unable to recollect something in more detail than this though =/

Reply:
Hmm, at this point in the Dockerfile we are running as `NB_USER`, so without the chown, `fix-permissions "${CONDA_DIR}"` will fail on my `.condarc`.

I think `--chown` is ~2 years old, so back then it would fail in many cases... but I hope that it's well-supported now?

I think the alternative would be something like
```Dockerfile
USER root
RUN chown blah
USER $NB_UID
```

A bit ugly, but if you're that concerned we could do it.

Reply:
I'm fine with what you want to try, but I want to warn about the vague memory I have that there isn't parity between COPY --chown and `RUN chown ...`  steps.